src/core/parser_internal/docgen_worker.hpp

```diff
@@ -0,0 +1,41 @@
+#pragma once
+
+#include "../symbol.hpp"
+#include "../token.hpp"
+#include "../json_writer.hpp"
+#include "../parse_worker.hpp"
+#include "../worker_routine.hpp"
+
+namespace docgen {
+namespace core {
+namespace parser_internal {
+
+using token_class_t = Token<Symbol>;
+
+using writer_class_t = JSONWriter<token_class_t>;
+
+using worker_class_t = ParseWorker<token_class_t, writer_class_t>;
```

Comment:
`worker_t`

Reply:
Not entirely possible. `ParseWorker` currently has `worker_t` defined, and all `SometingWorker`s inherit from `ParseWorker` and are within the `parser_internal` namespace, so this would lead to conflict. Same with `token_t`. I will instead adopt an `_internal_t` naming scheme for all of these.